src/js/playlist/playlist.js

```diff
@@ -39,15 +41,8 @@ define([
         return list;
     };
 
-    // A playlist item may have multiple different sources, but we want to stick with one.
-    var _filterSources = function(sources, providers, androidhls, itemDrm, preload) {
-
-        // legacy plugin support
-        if (!providers || !providers.choose) {
-            providers = new Providers({primary : providers ? 'flash' : null});
-        }
-
-        sources = _.compact(_.map(sources, function(originalSource) {
+    var _mapSources = function(sources, androidhls, itemDrm, preload) {
```

Comment:
In programming, a "map" is a term used to describe the application of a function against every element of a collection.

With a name like "mapSources" you really don't learn anything about what the function does, as a matter of fact, you may assume that it takes a function and returns the map of that function against sources. 

Which is of course false. 

Try to think of a better name for the function based on what it's purpose is, or what it actually does.